MakieCore/src/conversion.jl

```diff
@@ -30,3 +30,103 @@ conversion_trait(::Type{<: Heatmap}) = DiscreteSurface()
 
 struct VolumeLike <: ConversionTrait end
 conversion_trait(::Type{<: Volume}) = VolumeLike()
+
+
+"""
+    convert_arguments_typed(::Type{<: AbstractPlot}, args...)::NamedTuple
+
+Converts the arguments to their correct type for the Plot type.
+Throws apropriate errors if it can't convert!
+"""
+function convert_arguments_typed end
+
+
+function convert_arguments_typed(P::Type{<: AbstractPlot}, @nospecialize(args...))
+    convert_arguments_typed(conversion_trait(P), args...)
+end
+
+function convert_arguments_typed(::CT, @nospecialize(args...)) where {CT <: ConversionTrait}
+    return convert_arguments_typed(CT, args...)
+end
+
+function convert_arguments_typed(ct::Type{<: ConversionTrait}, @nospecialize(args...))
+    # We currently just fall back to not doing anything if there isn't a convert_arguments_typed defined for certain plot types.
+    # This makes `@convert_target` an optional feature right now.
```

Comment:
Explain here what the consequence is of args just being returned as is. Where will that lead to an error?

Reply:
Well, the consequence is, that it's optional, because you don't need to define it for every plot type this way.
The alternative would be erroring here, and demanding to add an `@convert_target` for the plot type.